dot/node.go

```diff
@@ -326,6 +326,7 @@ func newNodeC(cfg *Config, nn newNodeIface) (*Node, error) {
 
 	return node, nil
 }
+//go:generate mockgen -source=node.go -destination=mock_node.go -package=dot
```

Comment:
Change it to

```suggestion
//go:generate mockgen -destination=mock_node_test.go -package $GOPACKAGE . newNodeIface
```

So it's unexported to other packages (as a test file) and also using `source` isn't the best if it can be avoided. There are some corner cases where it's necessary (I don't remember which though haha), but it's better to just precise the interface name.

You should

Reply:
When I tried running this I get:
```
emack@system76-pc:~/projects/ChainSafe/gossamer/dot$ go generate
# command-line-arguments
./prog.go:27:38: cannot refer to unexported name dot.newNodeIface

```
so I needed to use the `-source` flag to solve issue.

Reply:
Totally! That was the case where you needed it 😄  Although you can use `$GOPACKAGE` instead of `dot` (if you want to)